velox/type/TimestampConversion.cpp

```diff
@@ -579,26 +588,24 @@ int64_t fromDateString(const char* str, size_t len) {
   return daysSinceEpoch;
 }
 
-int32_t
-castFromDateString(const char* str, size_t len, bool isNonStandardCast) {
+int32_t castFromDateString(const char* str, size_t len, bool isIso8601) {
   int64_t daysSinceEpoch;
   size_t pos = 0;
 
-  auto mode = isNonStandardCast ? ParseMode::kNonStandardCast
-                                : ParseMode::kStandardCast;
+  auto mode =
+      isIso8601 ? ParseMode::kStandardCast : ParseMode::kNonStandardCast;
   if (!tryParseDateString(str, len, pos, daysSinceEpoch, mode)) {
-    if (isNonStandardCast) {
+    if (isIso8601) {
       VELOX_USER_FAIL(
           "Unable to parse date value: \"{}\"."
-          "Valid date string patterns include "
-          "(YYYY, YYYY-MM, YYYY-MM-DD), and any pattern prefixed with [+-]",
+          "Valid date string pattern is (YYYY-MM-DD), "
+          "and can be prefixed with [+-]",
           std::string(str, len));
-
     } else {
       VELOX_USER_FAIL(
           "Unable to parse date value: \"{}\"."
-          "Valid date string pattern is (YYYY-MM-DD), "
-          "and can be prefixed with [+-]",
+          "Valid date string patterns include "
+          "(YYYY, YYYY-MM, YYYY-MM-DD), and any pattern prefixed with [+-]",
```

Comment:
nit: Can you include the full list of supported patterns here like you mentioned in configs.rst